packages/flutter/test/widgets/display_feature_sub_screen_test.dart

```diff
@@ -0,0 +1,218 @@
+    expect(renderBox.size.width, equals(800.0));
+    expect(renderBox.size.height, equals(290.0));
+    expect(renderBox.localToGlobal(Offset.zero), equals(const Offset(0,310)));
+  });
+
+  testWidgets('DisplayFeatureSubScreen anchorPoint multiple display features', (WidgetTester tester) async {
+    const Key childKey = Key('childKey');
+    final MediaQueryData mediaQuery = MediaQueryData.fromWindow(WidgetsBinding.instance!.window).copyWith(
+        displayFeatures: <DisplayFeature>[
+          const DisplayFeature(
+            bounds: Rect.fromLTRB(0, 290, 800, 310),
+            type: DisplayFeatureType.hinge,
+            state: DisplayFeatureState.unknown,
+          ),
+          const DisplayFeature(
+            bounds: Rect.fromLTRB(390, 0, 410, 600),
+            type: DisplayFeatureType.hinge,
+            state: DisplayFeatureState.unknown,
+          ),
+        ]
+    );
+
+    await tester.pumpWidget(
+      MediaQuery(
+        data: mediaQuery,
+        child: const DisplayFeatureSubScreen(
+          anchorPoint: Offset(1000, 1000),
+          child: SizedBox(
+            key: childKey,
+            width: double.infinity,
+            height: double.infinity,
+          ),
+        ),
+      ),
+    );
+
+    final RenderBox renderBox = tester.renderObject(find.byKey(childKey));
+    expect(renderBox.size.width, equals(390.0));
+    expect(renderBox.size.height, equals(290.0));
+    expect(renderBox.localToGlobal(Offset.zero), equals(const Offset(410,310)));
+  });
+
+  testWidgets('DisplayFeatureSubScreen anchorPoint non-splitting display features', (WidgetTester tester) async {
+    const Key childKey = Key('childKey');
+    final MediaQueryData mediaQuery = MediaQueryData.fromWindow(WidgetsBinding.instance!.window).copyWith(
+        displayFeatures: <DisplayFeature>[
+          // Top notch
+          const DisplayFeature(
+            bounds: Rect.fromLTRB(100, 0, 700, 100),
+            type: DisplayFeatureType.cutout,
+            state: DisplayFeatureState.unknown,
+          ),
+          // Bottom notch
+          const DisplayFeature(
+            bounds: Rect.fromLTRB(100, 500, 700, 600),
+            type: DisplayFeatureType.cutout,
+            state: DisplayFeatureState.unknown,
+          ),
+        ]
+    );
+
+    await tester.pumpWidget(
+      MediaQuery(
+        data: mediaQuery,
+        child: const DisplayFeatureSubScreen(
+          child: SizedBox(
+            key: childKey,
+            width: double.infinity,
+            height: double.infinity,
+          ),
+        ),
+      ),
+    );
+
+    // The display features provided are not wide enough to produce sub-screens
+    final RenderBox renderBox = tester.renderObject(find.byKey(childKey));
+    expect(renderBox.size.width, equals(800.0));
+    expect(renderBox.size.height, equals(600.0));
+    expect(renderBox.localToGlobal(Offset.zero), equals(Offset.zero));
+  });
+}
```

Comment:
None of these tests seem to exercise the code path with the Overlay?

Reply:
Ok, better tests and also changing how I calculate distances (I realised the current impl. is wrong) means I might have all these comments resolved tomorrow instead of today (as I had hoped). Just to keep you informed.